platform/mbed_assert.h

```diff
@@ -49,6 +49,65 @@ do {                                                     \
 } while (0)
 #endif
 
+
+#define _MBED_CONCAT_(a, b) a##b
+#define _MBED_CONCAT(a, b) _MBED_CONCAT_(a, b)
+#define _MBED_ASSERTION _MBED_CONCAT(_MBED_ASSERTION_AT_, __LINE__)
+
+/** MBED_STATIC_ASSERT
+ *  Declare compile-time assertions, results in compile-time error if condition is false
+ *
+ *  The assertion acts as a declaration that can be placed at file scope, in a
+ *  code block (except after a label), or as a member of a C++ class/struct/union.
+ *
+ *  @code
+ *  MBED_STATIC_ASSERT(MBED_LIBRARY_VERSION >= 120,
+ *          "The mbed library must be at least version 120");
+ *
+ *  int main() {
+ *      MBED_STATIC_ASSERT(sizeof(int) >= sizeof(char),
+ *              "An int must be larger than a char");
+ *  }
+ *  @endcode
+ */
+#ifdef __cplusplus
```

Comment:
When IAR is used, it is possible to delegate handling of static assertion to the extension `static_assert`.